core/arch/opnd.h

```diff
@@ -2465,6 +2497,14 @@ DR_API
 bool
 reg_is_strictly_ymm(reg_id_t reg);
 
+/**
+  * Assumes that \p reg is a DR_REG_ constant.
+  * Returns true if it refers to a zmm (512-bit multimedia) register.
+  * \deprecated Prefer reg_is_strictly_zmm().
+  */
+bool
+reg_is_zmm(reg_id_t reg);
```

Comment:
reg_is_ queries are deprecrated for simd, do not add new ones. Use reg_is_strictly_ instead.